### PR TITLE
Factor code out of UserCommander, remove circular dependency, testing

### PR DESCRIPTION
**Pull request by leogrim**
@andrewconner

### CODE REVIEW

**Comment by maxov** [comment]
:+1: :+1: :tada: :balloon:

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/emails/EmailTemplateProcessor.scala [hunk 1]
```diff
@@ -63,7 +64,7 @@ class EmailTemplateProcessorImpl @Inject() (
     libraryRepo: LibraryRepo,
     userRepo: UserRepo,
     keepRepo: KeepRepo,
-    userCommander: Provider[UserCommander],
+    s3ImageStore: S3ImageStore,
```
Comment (leogrim) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/emails/EmailTemplateProcessor.scala:67]: Victory match point.

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/UserEmailAddressCommander.scala [hunk 2]
```diff
@@ -136,7 +140,7 @@ class UserEmailAddressCommanderImpl @Inject() (db: Database,
 
     if (!force && isLast) Failure(new LastEmailAddressException(emailAddress))
     else if (!force && isLastVerified) Failure(new LastVerifiedEmailAddressException(emailAddress))
-    else if (!force && isPrimaryEmail(emailAddress)) Failure(new PrimaryEmailAddressException(emailAddress))
+    else if (!force && emailAddress.primary) Failure(new PrimaryEmailAddressException(emailAddress))
```
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/UserEmailAddressCommander.scala:143]: Can we call this `.isPrimary` instead of `.primary`?

Comment (leogrim) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/UserEmailAddressCommander.scala:143]: It's called 'primary' in the db for consistency with a field called `verified`, but yeah actually I was about to do that next :)

**Pull request merged**